src/test/scala/io/iohk/ethereum/blockchain/sync/StateSyncSpec.scala

```diff
@@ -121,7 +120,7 @@ class StateSyncSpec
         stateStorage = storages.stateStorage
       ) {
         override def mptStateSavedKeys(): Observable[Either[IterationError, ByteString]] = {
-          Observable.repeatEvalF(Task(Right(ByteString(1)))).takeWhile(_ => !loadingFinished)
+          Observable.repeat(Right(ByteString(1))).takeWhile(_ => !loadingFinished)
```

Comment:
Why not `Observable.interval(1.ms).map(_ => Right(ByteString(1)).takeWhile(_ => !loadingFinished)`?

I'm not very familiar with monix's internals, but I can imagine that just `repeat` gives not much time for other stuff on processing thread.